test/framework/src/main/java/org/elasticsearch/test/ESSingleNodeTestCase.java

```diff
@@ -455,6 +460,23 @@ protected void ensureNoInitializingShards() {
         assertFalse("timed out waiting for shards to initialize", actionGet.isTimedOut());
     }
 
+    /**
+     * waits until all free_context actions have been handled by the generic thread pool
+     */
+    protected void ensureAllFreeContextActionsAreConsumed() throws Exception {
+        NodeService nodeService = getInstanceFromNode(NodeService.class);
+        MockSearchTransportService searchTransportService = (MockSearchTransportService) nodeService.getSearchTransportService();
+        assertBusy(() -> {
```

Comment:
`assertBusy` can add quite some time to test execution. Could we use a `Semaphore` instead so we don't have to busy-wait?

Reply:
This has now been removed, as we'll simply check using the list tasks API